api/v1/ptpconfig_types.go

```diff
@@ -61,11 +61,12 @@ type PtpConfigList struct {
 }
 
 type PtpProfile struct {
-	Name        *string `json:"name"`
-	Interface   *string `json:"interface,omitempty"`
-	Ptp4lOpts   *string `json:"ptp4lOpts,omitempty"`
-	Phc2sysOpts *string `json:"phc2sysOpts,omitempty"`
-	Ptp4lConf   *string `json:"ptp4lConf,omitempty"`
+	Name            *string `json:"name"`
+	Interface       *string `json:"interface,omitempty"`
+	Ptp4lOpts       *string `json:"ptp4lOpts,omitempty"`
+	Phc2sysOpts     *string `json:"phc2sysOpts,omitempty"`
+	BoundaryClockHA *bool   `json:"boundaryClockHA,omitempty"`
```

Comment:
I think we should call it as EnableHA and the logic decided if config  has HA capability
It is not HA for boundary clock, it is more like HA between two PTP enabled nics  which could also apply to ordinary clock  right ?
May  be  if you create a Struct 
"HighAvailability": {
      "EnableHA": true|false,
      "PreferredPrimary": true|false,   // each nic will have initial state  request and status will  inform which nic is primary
      }
      Remember each nic has its own ptpConfig, in that case both has to have this section . 

     Probably best will be to introduce new  ptpHAConfig  and introduce controller to update HA preferences
monitor changes and update  linuxPtpDaemon of new change.
ptpHAConfig {
  "EnableHA": true|false,
  "HAPTPProfiles":
	- profile1
	- profile2 
  "PreferredPrimay": profile1
  "HeartBeatTimeOut": int
}

Reply:
+1 from my side. I'll modify this PR and rescope it so it matches the proposed change, and not only the field.

Reply:
Would it make sense to have a HAPtPProfile? IIUC they would be just the same as the PTPProfiles, tbh. Anyway, I'll put my changes on the PtpConfig, let's iterate afterwards

Reply:
Like, explaning myself, this is the sample for a multiple nic config:
```
apiVersion: ptp.openshift.io/v1
kind: PtpConfig
metadata:
  name: boundary-clock-ptpconfig
  namespace: ptp
spec:
  profile:
  - name: "profile1"
    ptp4lOpts: "-s -2"
    phc2sysOpts: "-a -r"
    ptp4lConf: |
      [ens7f0]
      masterOnly 0
      [ens7f1]
      masterOnly 1
      [ens7f2]
      masterOnly 1
  recommend:
  - profile: "profile1"
    priority: 4
    match:
    - nodeLabel: "node-role.kubernetes.io/worker"
```
I would just add the EnableHA and the different options there. Everything else should be the same, wdyt?